docs/release-management.md

```diff
@@ -0,0 +1,139 @@
+# Proposal for release management in the submariner-io project
+
+## Introduction
+
+Releasing submariner consists of several manual [steps](https://submariner-io.github.io/contributing/release-process/)
+that need to be carefully executed in order across multiple projects.
+
+This process can be automated as described in the following sections.
+
+## What's a release
+
+   A submariner release is the sum of all components, tagged on a specific set of versions,
+   that work and are tested together.
+   
+   The components include:
+   
+   * subctl
+   * submariner-operator
+   * submariner core (submariner engine, route agent, globalnet ipam)
+   * lighthouse (lighthouse agent, lighthouse coredns)
```

Comment:
Do you want to have Shipyard on the list? You mention it later so it can be listed here with a note that it's the base env for building the other projects.